From 1527fb44c331f2c61224514beb758895a5f85c8d Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Thu, 19 Mar 2015 07:51:48 +0100
Subject: Revert various changes conflicting with Maarten Lankhorst's gstreamer
 hack v5.

Note: These changes have to be reapplied with the corresponding -post patchset
afterwards.
---
 dlls/ntdll/thread.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/dlls/ntdll/thread.c b/dlls/ntdll/thread.c
index 30a4720..843be0f 100644
--- a/dlls/ntdll/thread.c
+++ b/dlls/ntdll/thread.c
@@ -385,8 +385,6 @@ void terminate_thread( int status )
 void exit_thread( int status )
 {
     static void *prev_teb;
-    shmlocal_t *shmlocal;
-    sigset_t sigset;
     TEB *teb;
 
     if (status)  /* send the exit code to the server (0 is already the default) */
@@ -425,11 +423,6 @@ void exit_thread( int status )
         }
     }
 
-    sigemptyset( &sigset );
-    sigaddset( &sigset, SIGQUIT );
-    pthread_sigmask( SIG_BLOCK, &sigset, NULL );
-    if (interlocked_xchg_add( &nb_threads, -1 ) <= 1) _exit( status );
-
     close( ntdll_get_thread_data()->wait_fd[0] );
     close( ntdll_get_thread_data()->wait_fd[1] );
     close( ntdll_get_thread_data()->reply_fd );
-- 
2.3.2

